### ISSUE TITLE
Improvement ideas

### ISSUE THREAD

**Issue by davidbossanyi** [davidbossanyi/hill-of-towie-wind-turbine-power-prediction#13]
Use `catboost` outside of `AutoML` to give better control and monitoring

```python
from catboost import CatBoostRegressor


model = CatBoostRegressor(
    iterations=10000,
    learning_rate=0.05,  # Lower learning rate for more stable training
    early_stopping_rounds=200,  # More patience
    eval_metric='MAE',
    use_best_model=True,
    verbose=100,  # Print every 100 iterations
    l2_leaf_reg=3.0,
    max_depth=6,
    random_seed=42
)

# Fit with validation set for monitoring?
model.fit(
    X_train.to_pandas(), 
    y_train.to_pandas(),
    eval_set=(X_validation.to_pandas(), y_validation.to_pandas()),
    verbose_eval=100,
    plot=True
)

print(f"Best iteration: {model.get_best_iteration()}")
print(f"Best score: {model.get_best_score()}")
```

Could use [`randomized_search`](https://catboost.ai/docs/en/concepts/python-reference_catboost_randomized_search) from `catboost` itself or even better, [`optuna` with pruning](https://github.com/optuna/optuna-examples/blob/main/catboost/catboost_pruning.py) to optimize the hyperparameters a bit. In particular, learning rate, l2 leaf reg and depth, and perhaps others too.

* Better feature pruning, explore some other features
* Look at the distribution of residuals to identify where the model is lacking, and to guide feature engineering
